lib/edools/lesson_progress.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
+module Edools
+  class LessonProgress
+    attr_reader :lesson_progress, :paginate, :errors
+
+    def initialize(data = {})
+      @lesson_progress = objectify_lesson_progress(data)
+      @errors = data.dig(:errors)
+    end
+
+    def self.find_paginated_from_enrollments(enrollment_id: , page: 1, per_page: 50)
+      raise ArgumentError, 'missing enrollment_id' if enrollment_id.nil?
+
+      url = "#{base_url_enrollments}/#{enrollment_id}/lessons_progresses"
+      data = { enrollment_id: enrollment_id, page: page, per_page: per_page }
+
+      response = Edools::ApiRequest.request(:get, url, data)
+
+      OpenStruct.new(
+        lessons_progresses: objectify_lessons_progresses(response[:lessons_progresses]),
+        paginate: objectify_paginate(response)
+      )
+    end
+
+    def self.find_paginated_from_students(student_id: , page: 1, per_page: 50)
+      raise ArgumentError, 'missing student_id' if student_id.nil?
+
+      url = "#{base_url_students}/#{student_id}/lessons_progresses"
+      data = { student_id: student_id, page: page, per_page: per_page }
+
+      response = Edools::ApiRequest.request(:get, url, data)
+
+      OpenStruct.new(
+        lessons_progresses: objectify_lessons_progresses(response[:lessons_progresses]),
+        paginate: objectify_paginate(response)
+      )
+    end
+
+    def self.find_paginated_from_school_products(school_product_id: , page: 1, per_page: 50)
+      raise ArgumentError, 'missing school_product_id' if school_product_id.nil?
+
+      url = "#{base_url_school_products}/#{school_product_id}/lessons_progresses"
+      data = { school_product_id: school_product_id, page: page, per_page: per_page }
+
+      response = Edools::ApiRequest.request(:get, url, data)
+
+      OpenStruct.new(
+        lessons_progresses: objectify_lessons_progresses(response[:lessons_progresses]),
+        paginate: objectify_paginate(response)
+      )
+    end
+
+    def self.show(data = {})
+      raise ArgumentError, 'missing id' unless data.key? :id
+
+      response = Edools::ApiRequest.request(:get, base_url, data)
+
+      response[:lessons_progresses].map do |lesson_progress|
+        new lesson_progress
+      end
+    end
+
+    def self.base_url_enrollments
+      "#{Edools.base_url}/enrollments"
+    end
+
+    def self.base_url_students
+      "#{Edools.base_url}/students"
+    end
+
+    def self.base_url_school_products
+      "#{Edools.base_url}/school_products"
+    end
+
+    def self.base_url
+      "#{Edools.base_url}/lessons_progresses"
+    end
+
+    private
+
+    def self.objectify_paginate(paginate)
+      return paginate if paginate.nil?
```

Comment:
When paginate is going to be `nil`? If that happens when there's only a single page I imagine users may rely on this pagination information in order to avoid multiples calls, so maybe we can generate a structure like the one below just to keep the standard?

something like: 
```
{
  current_page: 1,
  per_page: lessons_progresses.size,
  total_pages: 1,
  total_count: lessons_progresses.size
}
```

Does it make sense? I fell like this is a much more expected behaviour although being able to `unless foobar.paginate` seems cool too.

Reply:
I prevent any possible problem that the pagination will have, so I do this nil validation.

I'm more comfortable passing the values coming from the API